test/util/branch-selector.html

```diff
@@ -0,0 +1,82 @@
+<!doctype html>
+<html>
+<head>
+  <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
+  <title>branch-selector-tests</title>
+
+  <script src="../../src/bower_components/webcomponentsjs/webcomponents-lite.min.js"></script>
+  <script src="../../src/bower_components/web-component-tester/browser.js"></script>
+  <script src="../../src/bower_components/iron-test-helpers/mock-interactions.js"></script>
+
+  <!-- Import the element to test -->
+  <link rel="import" href="../../src/util/branch-selector.html">
+
+</head>
+<body>
+
+  <test-fixture id="branch-selector-fixture">
+    <template>
+      <branch-selector></branch-selector>
+    </template>
+  </test-fixture>
+
+  <script>
+  suite('<branch-selector>', function() {
+    var branchSelector;
+
+    setup(function() {
+    branchSelector = fixture('branch-selector-fixture');
+    });
+
+    test('on focus show dropdown', function() {
+      MockInteractions.focus(branchSelector);
+      assert.isFalse(branchSelector.hideSelectionMenu);
+    }),
+    
+    test('on blur hide dropdown', function() {
+      MockInteractions.focus(branchSelector);
+      MockInteractions.blur(branchSelector);
+      assert.isTrue(branchSelector.hideSelectionMenu);
+    }),
+
+    test('selecting branch', function() {
+      branchSelector.branches = ["test","testing"];
+      Polymer.dom(branchSelector.root).querySelector('iron-selector').select('test');
+      assert.equal(branchSelector.selected, "test");
+    }),
+
+    test('selecting branch hide dropdown', function(done) {
+      branchSelector.branches = ["test","testing"];
+      MockInteractions.focus(branchSelector);
+      var ironSelector = Polymer.dom(branchSelector.root).querySelector('iron-selector');
+      ironSelector.select('test');
+      flush(function() {
+        MockInteractions.tap(ironSelector.selectedItem);
+        assert.isTrue(branchSelector.hideSelectionMenu);
+        done();
+      });
+    }),
+
+    test('filtering branches sets value', function(done) {
+      branchSelector.branches = ["test","testing"];
+      var input = Polymer.dom(branchSelector.root).querySelector('paper-input')
+      input.value = "ing";
+      assert.equal(branchSelector.filter,input.value);
+    }),
+
+    test('filtering branches hides items', function(done) {
+      branchSelector.branches = ["test","testing"];
+      var input = Polymer.dom(branchSelector.root).querySelector('paper-input')
+      input.value = "ing";
+      var ironSelector = Polymer.dom(branchSelector.root).querySelector('iron-selector');
```

Comment:
Instead of poking inside the element's local DOM, would it not be possible to just change the `branchSelector.branch` attribute in order to select a branch?